app/src/app/services/connectors/authentication/authentication.service.ts

```diff
@@ -102,6 +102,47 @@ export class AuthenticationService extends ApiConnector {
         )
     }
 
+    /**
+     * User register sequence
+     * @returns of the logged in user account
+     */
+    public register() {
+        return this.getAuthType().pipe(
+            // retrieve authentication configuration
+            concatMap(authnType => {
+                if (authnType == "oidc") {
+                    let url = `${this.baseUrl}/authn/${authnType}/login`;
+                    // oidc login
+                    url += `?destination=${encodeURIComponent(window.location.href+'register')}`;
+                    // redirect to OIDC Identity Provider
+                    window.location.href = url;
+                }
+                return [].map(() => ({next: () => true}));
+            }),
+            catchError(this.handleError_raise<UserAccount>()),
+            share()
+        );
+    }
+
+    public handleRegisterRedirect(): Observable<any> {
+        return this.getAuthType().pipe(
+            // retrieve authentication configuration
+            concatMap(authnType => {
+                if (authnType == "oidc") {
+                    return this.http.post(`${this.baseUrl}/user-accounts/register`, {responseType: 'text'}).pipe(
+                        map(res => {
+                            this.currentUser = undefined;
```

Comment:
A couple issues arise with setting the current user to `undefined` here. 
- Technically, the user is still recognized as being logged in on the REST API side. It's correct that the user should not be able to login, since their account is still `pending`, but this functionality should be handled in the `checkStatus()` function that's called in `app.component.ts` after this function returns. This will log the user out of the REST API as well
- The `checkStatus()` also displays alert information based on the user account status (i.e. successful registration), so when we unset the current user, this information is not displayed. The user session needs to be retrieved in this function in order for that to work as intended